### ISSUE TITLE
Have `no-empty` also handle block with useless label and unused expression

### ISSUE THREAD

**Issue by brettz9** [eslint/eslint#5518]
**What version are you using?**

`master` for `2.3.0`.

**What did you do?**

Use the `no-empty` rule against:

```js
{
myLabel: anUnusedExpression
}
```

**What happened?**

The structure is not flagged as violating the rule.

**What did you expect to happen?**

I think this rule should flag the structure as a problem since it is not, or should not, be caught by `no-unused-expressions` and since such structures are useless and ought to be catchable.

(As a side note, I think a bare `{}` ought to be added to the `no-empty` docs as another example of a problematic empty block, especially since some might confuse it with an expression (object literal) and thus think it wasn't handled here.)

**Comment by ilyavolodin** [comment]
This should be flagged by `[no-extra-label](http://eslint.org/docs/rules/no-extra-label)` rule. We usually try to stay away from rules flagging the same pattern (it would be very complicated to auto-fix those rules).

**Comment by brettz9** [comment]
Ah yes, I hadn't been granular enough--I see my concerns are met by `no-extra-label`, as you say, as well as `no-lone-blocks`.

One question though--does `non-lone-blocks` avoid flagging blocks with labels? If someone had something like this:

```js
longBlock: {
    if (condition) {
        // ...
    }
    else if (someCondition) {
        break longBlock;
    }

    // More code
}
```

...the block actually serves a purpose and I think should not be flagged.

**Comment by ilyavolodin** [comment]
I'm pretty sure that `no-lone-blocks` will only flag blocks that don't have any content in them. Even putting a comment inside will make the rule ignore a block.

**Comment by mysticatea** [comment]
`no-lone-blocks` doesn't warn labeled blocks.

https://github.com/eslint/eslint/blob/master/lib/rules/no-lone-blocks.js#L39
Because `parent.type` is `"LabelledStatement"`.

**Comment by brettz9** [comment]
ok thank you, @ilyavolodin and @mysticatea

**Comment by brettz9** [comment]
Hmm...I'm getting "Block is redundant: no-lone-blocks" on the [demo](http://eslint.org/demo/) for this code:

```js
var abc = 5;

{
    myLabel: abc
}
```

**Comment by mysticatea** [comment]
That's correct.

**Comment by brettz9** [comment]
I see--I'm confusing the outer block with the labeled block, sorry.
